### PR TITLE
Call settings updates

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Links

_Provide all JIRA tickets and/or GitHub issues related to this PR, if applicable._

### 🎯 Goal

_Describe why we are making this change._

### 📝 Summary

_Provide bullet points with the most important changes in the codebase._

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

|  Before  |  After  |
| -------- | ------- |
|  `img`   |  `img`  |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: Sources/StreamVideo/CallSettings/CallSettingsManager.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+public enum CallSettingsStatus: String {
+    case enabled
+    case disabled
+    
+    var next: Self {
+        self == .enabled ? .disabled : .enabled
+    }
+    
+    var toBool: Bool {
```
Comment (ipavlidakis) [inline Sources/StreamVideo/CallSettings/CallSettingsManager.swift:52]:
```suggestion
    var boolValue: Bool {
```

File: Sources/StreamVideo/CallSettings/CameraManager.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    /// Toggles the camera state.
+    public func toggle() async throws {
+        try await updateVideoStatus(status.next)
+    }
+    
+    /// Flips the camera (front to back and vice versa).
+    public func flip() async throws {
+        let next = direction.next()
+        await withCheckedContinuation { [unowned self] continuation in
```
Comment (ipavlidakis) [inline Sources/StreamVideo/CallSettings/CameraManager.swift:33]: Why not `weak self`?

File: Sources/StreamVideo/CallSettings/SpeakerManager.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+public class SpeakerManager: ObservableObject, CallSettingsManager {
+        
+    internal let callController: CallController
+    @Published public internal(set) var status: CallSettingsStatus
+    @Published public internal(set) var audioOutputStatus: CallSettingsStatus
+    internal let state = CallSettingsState()
+    
+    init(
+        callController: CallController,
+        initialStatus: CallSettingsStatus,
```
Comment (ipavlidakis) [inline Sources/StreamVideo/CallSettings/SpeakerManager.swift:17]: `initialStatus` and `audioOutputStatus` are not clear to me what to me. What about `initialSpeakerStatus` and initial `initialAudioOutputStatus`?

File: Sources/StreamVideo/Utils/SystemEnvironment.swift [hunk 4]
```diff
@@ -46,20 +46,4 @@ enum SystemEnvironment {
     private static var hasEmbeddedMobileProvision: Bool {
         Bundle.main.path(forResource: "embedded", ofType: "mobileprovision") != nil
     }
-
-    static var isExtention: Bool {
```
Comment (ipavlidakis) [inline Sources/StreamVideo/Utils/SystemEnvironment.swift:50]: Why are we removing those?

File: Sources/StreamVideo/CallSettings/CameraManager.swift (hunk 2 above)
Comment (martinmitrevski) [reply Sources/StreamVideo/CallSettings/CameraManager.swift:33]: you are right, better to use weak here.

File: Sources/StreamVideo/CallSettings/SpeakerManager.swift (hunk 3 above)
Comment (martinmitrevski) [reply Sources/StreamVideo/CallSettings/SpeakerManager.swift:17]: changed

File: Sources/StreamVideo/Utils/SystemEnvironment.swift (hunk 4 above)
Comment (martinmitrevski) [reply Sources/StreamVideo/Utils/SystemEnvironment.swift:50]: these are not used anywhere

**Review by ipavlidakis — APPROVED** [review]

**Pull request merged**